### DISCUSSION TITLE
Help please. VSC has betrayed me.

### DISCUSSION THREAD

**Discussion by Jyers** [Jyers/Marlin#723] [General Discussion]
I don't even know what to say. I was trying to pull from upstream, but now I can't do anything without just about everything throwing errors. I've tried a fresh clone, reinstalling visual studio code, nothing seems to work. I can't do much with it acting like this so I figured I'd reach out to see if anyone has seen this before or can think of an issue. I'm certain its something with VSC I just don't know what...

![2021-05-06 18 01 43](https://user-images.githubusercontent.com/76993396/117371396-d1e8ff00-ae95-11eb-9efc-2ea5241209ec.png)
![2021-05-06 18 02 48](https://user-images.githubusercontent.com/76993396/117371401-d44b5900-ae95-11eb-8bad-05118d8edeae.png)

**Comment by adam3654** [comment]
That happened to me when I first set up the workspace. I remember googling the issue and it said it was a known bug, but it went away for me. I don't remember if I did anything to fix it, but it always compiled okay.

**Comment by Cal-LF** [comment]
![Squiggles](https://user-images.githubusercontent.com/76499230/117375416-ec72a680-ae9c-11eb-9c05-04660bd7d2bd.jpg)
I've been getting Squiggly Line errors since VSC updated a couple weeks ago.
Ctl/Shft/P gets to a list that turns off these errors but only for the current session.

**Comment by Jyers** [comment]
One update is that this issue seems to step from the .vscode folder not being generated when loading my repo in vsc. I have yet to figure out any way to get it to generate this, but it seems to do so when cloning the main marlin repo, so I don't know what's up

**Comment by Ka8oum** [comment]
Hi, 
at some point in time I just couldn't compile your firmware and I had a lot of errors, then I noticed the default folder from platformio was not selected correctly in the bottom bar, I don't know if you have the same thing but at least it was the issue for me, now I make sure to check it regularly

![image](https://user-images.githubusercontent.com/83807872/117412564-37c89b80-af15-11eb-897a-d6c0f5b2b66e.png)

![image](https://user-images.githubusercontent.com/83807872/117412789-79f1dd00-af15-11eb-815b-f65fddfb94d8.png)

**Comment by tiagofreire-pt** [comment]
@Jyers , can you compile successfully the vanilla Marlin bugfix repo?

**Comment by Cal-LF** [comment]
I have to move the folder to a short path directory like E:\temp or a long path causes problems.

**Comment by Reggi3** [comment]
Did you uninstall or did you do a complete removal and uninstall?  If you're on windows, If it was just an uninstall, the following folders should apparently be removed to perform a full cleanup before reinstall, although I'm not sure if there would be any junk lurking in the registry:
```
%AppData%\Code 
%userprofile%\.vscode
```
I'm not sure of the wider impact to your setup but I can imagine it to be fairly severe, although I'm sure you'll be mindful of that.

**Reply by Timbo200** [reply]
Concur!

As a complete noob I completely borked my initial install of (release) VSC.  Reinstall and all manner of deletions did nothing to to fix.

Solution I had was uninstall and nuke the "code" directory and everything under it.  A regular uninstall in Win 10 left this intact.

> Depending on your platform, the user settings file is located here:
> 
> Windows %APPDATA%\Code\User\settings.json
> macOS $HOME/Library/Application Support/Code/User/settings.json
> Linux $HOME/.config/Code/User/settings.json
> 

https://code.visualstudio.com/docs/getstarted/settings

**Comment by smooreace** [comment]
Ive put out a call to everyone I know that might have some background. If I
find anyone that might can help I will send them here

**Comment by Jyers** [comment]
I've made progress. Maybe.... Here's where I'm at now, this appears on most files I try to edit:

![20210507-120849279](https://user-images.githubusercontent.com/76993396/117478348-08258d80-af2d-11eb-9f9c-4fec08344ba2.png)

**Reply by tiagofreire-pt** [reply]
I have the same issue here. It seems now requires some new package installer.

Regardless, it compiles flawlessly.

**Comment by smooreace** [comment]
[image: image.png]

**Comment by Jyers** [comment]
Okay, so I have definitely nailed down what the issue is:

This is what the bottom right of my lower nav bar look like
![20210507-162306753](https://user-images.githubusercontent.com/76993396/117504894-2ac99d80-af51-11eb-9cb9-224ac1916826.png)
This is what it should display
![20210507-162243609](https://user-images.githubusercontent.com/76993396/117504882-2604e980-af51-11eb-9408-536160166e3b.png)

It seems VSC isn't using the correct c/c++ configuration. It's supposed to get this from the pio.ini file, but something must be wrong. This is a good start, but I still have yet to find a solution.

**Reply by Jyers** [reply]
I've tried that. I know platform IO wants c++11/c11, but setting it to that doesn't seem to fix the issue. This is getting mildly infuriating at this point.

**Comment by Jyers** [comment]
Okay, so I hacked together a fix. In the end it was definitely the .vscode folder that was causing issues. The main one being that for whatever reason vscode wouldn't generate the folder when I opened my repo.

I was finally able to fix it by cloning the main marlin repo, putting my platformio.ini file into it, then opening it with vscode. Because vscode doesn't have the same problem with the main repo it generated a .vscode folder just fine. I then copied that folder into my repo and modified the includes to point to the right folder.

I would still really like to know why vsc was refusing to generate a .vscode folder for my repo, so if anyone has any ideas I would love to hear them...
